webui/react/src/components/InlineTextEdit.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { KeyboardEvent, useCallback, useRef, useState } from 'react';
+import ContentEditable from 'react-contenteditable';
+
+import Icon from 'components/Icon';
+import { IndicatorUnpositioned } from 'components/Spinner';
+
+import css from './InlineTextEdit.module.scss';
+
+interface Props {
+  setValue: (newValue: string) => void;
+  value: string;
+}
+
+const InlineTextEdit: React.FC<Props> = ({ setValue, value }: Props) => {
+  const [ isChanged, setIsChanged ] = useState(false);
+  const [ isFocused, setIsFocused ] = useState(false);
+  const [ isSaving, setIsSaving ] = useState(false);
+  const innerValueRef = useRef<string>(value);
+  const inputRef = React.useRef<HTMLElement>();
+
+  const clear = useCallback(() => {
+    innerValueRef.current = value;
+    setIsChanged(false);
+
+    if (!inputRef.current) return;
+    inputRef.current.blur();
+  }, [ value ]);
+  const focus = useCallback(() => {
+    if (!inputRef.current) return;
+    inputRef.current.focus();
+  }, []);
+  const save = useCallback(async () => {
+    setIsSaving(true);
+    setIsChanged(false);
+    await setValue(innerValueRef.current);
```

Comment:
blocking: could we also handle request (setValue) failures as well? If my request fails this seems to stay in the spinning state.

Reply:
added, nice catch.